src/App.js

```diff
@@ -1,14 +1,31 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ErrorBoundary from './hoc/ErrorBoundary/ErrorBoundary';
 import Header from './views/Header/Header';
+import MovieDetails from './views/MovieDetails/MovieDetails';
 import MainContent from './views/MainContent/MainContent';
 import Footer from './views/Footer/Footer';
 
 export default function App() {
+  const [showMovieDetails, setShowMovieDetails] = useState(false);
+  const [movie, setMovie] = useState({});
+
+  const onMovieClick = (movie) => {
+    setShowMovieDetails(true);
+    setMovie(movie);
+    window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+  };
+
   return (
     <ErrorBoundary>
-      <Header />
-      <MainContent />
+      {showMovieDetails
+        ? (
+          <MovieDetails
```

Comment:
We can also put this components inside header and make conditional rendering there

Reply:
This was my first intention, but it seemed that it looked less readable and more overloaded.

Updated (moved MovieDetails inside Header).